terraform/api.tf

```diff
@@ -21,7 +21,7 @@ module "api" {
   django_cors_origin_regex_whitelist = ["^https:\\/\\/[0-9a-z\\-]+--gui-dandiarchive-org\\.netlify\\.app$"]
 
   additional_django_vars = {
-    DJANGO_DANDI_DANDISETS_BUCKET_NAME = aws_s3_bucket.api_dandisets_bucket.id
+    DJANGO_DANDI_DANDISETS_BUCKET_NAME = aws_s3_bucket.sponsored_bucket.id
```

Comment:
Could/should policy below restrict to what key prefix(es) this instance could have write access to?

Reply:
I'm 90% sure it can, but this PR is about switching the bucket, so we can tackle that later.